src/solvers/dg_simplices/fluxdiff.jl

```diff
@@ -0,0 +1,182 @@
+"""
+  function hadamard_sum_ATr!(du, ATr, volume_flux, u, skip_index=(i,j)->false)
+
+Computes the flux difference ∑_j Aij * f(u_i, u_j) and accumulates the result into `du`. 
+- `du`, `u` are vectors
+- `ATr` is the transpose of the flux differencing matrix `A`. The transpose is used for 
+faster traversal since matrices are column major in Julia. 
+"""
+@inline @muladd function hadamard_sum_ATr!(du, ATr, volume_flux::VF, u, skip_index=(i,j)->false) where {VF}
+  rows,cols = axes(ATr)
+  for i in cols
+    ui = u[i]
+    val_i = du[i]
+    for j in rows
+      if !skip_index(i,j)                
+          val_i += ATr[j,i] * volume_flux(ui, u[j])
+      end
+    end
+    du[i] = val_i 
+  end
+end
+
+function build_lazy_physical_derivative(elem::Int, orientation::Int, mesh::VertexMappedMesh{2, Tri}, dg, cache) 
+  @unpack Qrst_skew_Tr = cache
+  @unpack rxJ, sxJ, ryJ, syJ = mesh.md
+  QrskewTr, QsskewTr = Qrst_skew_Tr
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,elem] * QrskewTr + sxJ[1,elem] * QsskewTr))
+  else
+    return LazyArray(@~ @. 2 * (ryJ[1,elem] * QrskewTr + syJ[1,elem] * QsskewTr))
+  end
+end
+
+function build_lazy_physical_derivative(elem::Int, orientation::Int, mesh::VertexMappedMesh{3, Tet}, dg, cache) 
+  @unpack Qrst_skew_Tr = cache
+  QrskewTr, QsskewTr, QtskewTr = Qrst_skew_Tr
+  @unpack rxJ, sxJ, txJ, ryJ, syJ, tyJ, rzJ, szJ, tzJ = mesh.md
+  if orientation == 1
+    return LazyArray(@~ @. 2 * (rxJ[1,elem]*QrskewTr + sxJ[1,elem]*QsskewTr + txJ[1,elem]*QtskewTr))
+  elseif orientation == 2
+    return LazyArray(@~ @. 2 * (ryJ[1,elem]*QrskewTr + syJ[1,elem]*QsskewTr + tyJ[1,elem]*QtskewTr))
+  elseif orientation == 3
+    return LazyArray(@~ @. 2 * (rzJ[1,elem]*QrskewTr + szJ[1,elem]*QsskewTr + tzJ[1,elem]*QtskewTr))
+  end
```

Comment:
What's the advantage of `LazyArray`s here? Could you please explain it in a (brief) comment?

Reply:
Yup! Adding the following:

> For DGMulti implementations, we construct "physical" differentiation operators by taking linear combinations of reference differentiation operators scaled by geometric change of variables terms. We use LazyArrays.jl for lazy evaluation of physical differentiation operators, so that we can compute linear combinations of differentiation operators on-the-fly in an allocation-free manner.

Reply:
Great, that's nice and helps people to understand why the code does what it does :+1: